apps/emqx_durable_storage/src/emqx_ds_replication_layer.erl

```diff
@@ -135,6 +141,20 @@ add_generation(DB) ->
 update_db_config(DB, CreateOpts) ->
     emqx_ds_replication_layer_meta:update_db_config(DB, CreateOpts).
 
+-spec list_generations_with_lifetimes(emqx_ds:db()) ->
+    #{generation_rank() => emqx_ds:generation_info()}.
+list_generations_with_lifetimes(DB) ->
+    Nodes = list_nodes(),
```

Comment:
I think the RPC should be targeted towards the shard leader. Once we implement HA, different nodes will contain overlapping sets of generations.

Reply:
Thanks, fixed.